### PR TITLE
test: move to koji-osbuild-tests package

### PR DESCRIPTION
**Pull request by larskarlitski**
This is similar to how other osbuild packages are testing: everything
that's needed for testing is included in the tests package or a
dependency of it. The test runner then runs every executable in
/usr/libexec/tests/<packagename>. This gives a simple test API to
projects depending on this package (notably osbuild-composer).

Note that because all tests are now installed, this commit breaks using
the same containers for rapid iteration local testing. Combining the two
use-cases was unfeasible for the time being. Tests can still be run
locally via building the package and using something like
osbuild-composer's tools/deploy-qemu (though with a larger iteration
time).

### CODE REVIEW

**Review by teg — DISMISSED** [review]
Nice!

File: test/container/builder/run-kojid.sh [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
 set -ux
 
-if ls /share/rpms/*.rpm 1> /dev/null 2>&1; then
-  echo "Using RPMs"
-  rm /usr/lib/koji-builder-plugins/osbuild.py
-  rpm -i /share/rpms/koji-osbuild-?-0.*.rpm \
-         /share/rpms/koji-osbuild-builder-*.rpm
-fi
+rpm -i /share/rpms/koji-osbuild-?-0.*.rpm \
+       /share/rpms/koji-osbuild-builder-*.rpm
```
Comment (gicmo) [inline test/container/builder/run-kojid.sh:5]: What if we kept the `if`, but removed the `rm`. Then add an `elif ls` to see if `/share/plugins/builder/osbuild.py` exists and if so copy that over.

**Review by gicmo — DISMISSED** [review]
:heart: This is amazing!! Love this so much!

I think with a few tweaks we could support the local dev workflow. Besides the inline comments, we would just have to add a `local` command line option for `build-container.sh` and `copy-creds.sh` that would change the path to either local or `/usr/share/`. And `build-container.sh` would also copy the plugins to the share dir, where they could be picked up by the modified entry point scripts. Happy to have that also in a follow up though!

**Comment by larskarlitski** [comment]
> I think with a few tweaks we could support the local dev workflow.

You are absolutely right of course. I've changed it according to your suggestions and explained it differently in the commit. This makes it even simpler :)

**Review by gicmo — APPROVED** [review]
Even better now!

**Pull request merged**